src/roiextractors/extractors/tiffimagingextractors/scanimagetiff_utils.py

```diff
@@ -119,18 +125,38 @@ def parse_metadata(metadata: dict) -> dict:
     active_channels = parse_matlab_vector(metadata["SI.hChannels.channelsActive"])
     channel_indices = np.array(active_channels) - 1  # Account for MATLAB indexing
     channel_names = np.array(metadata["SI.hChannels.channelName"].split("'")[1::2])
+    scan_line_rate = 1 / float(metadata["SI.hRoiManager.linePeriod"])
     channel_names = channel_names[channel_indices].tolist()
     num_channels = len(channel_names)
-    if "RoiGroups" in metadata.keys():
-        roi_metadata = metadata["RoiGroups"]
-    else:
-        roi_metadata = None
+    roi_metadata = metadata["RoiGroups"]
+    grid_spacing = None
+    grid_spacing_unit = "n.a"
+    origin_coords = None
+    origin_coords_unit = "n.a"
+    try:
```

Comment:
this would probably be cleaner with a if block if the nested scanfields attr is in roi_metadata